### ISSUE TITLE
OutOfRange exception in PreVisit(CallNode node), in LSP

### ISSUE THREAD

**Issue by MikeStall** [microsoft/Power-Fx#2759]
We're seeing this in our logs:

Just started today 

System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')
   at System.SZArrayHelper.get_Item[T](Int32 index)
   at Microsoft.PowerFx.Core.Binding.TexlBinding.Visitor.PreVisit(CallNode node)
   at Microsoft.PowerFx.Engine.ComputeBinding(CheckResult result)
   at Microsoft.PowerFx.CheckResult.ApplyBindingInternal()
   at Microsoft.PowerFx.LanguageServerProtocol.Handlers.LanguageServerOperationContext.ExecuteHostTaskAsync(String uri, Action`1 task, CancellationToken cancellationToken)
   at Microsoft.PowerFx.LanguageServerProtocol.Handlers.OnDidChangeLanguageServerNotificationHandler.HandleAsync(LanguageServerOperationContext operationContext, CancellationToken cancellationToken)
   at Microsoft.PowerFx.LanguageServerProtocol.LanguageServer.OnDataReceivedAsyncInternal(LanguageServerInput input, CancellationToken cancellationToken)

**Comment by MikeStall** [comment]
I _think_ this was happening while I was authoring the following expression in playground:  (not entirely sure)


```
Join(Sequence(3), ForAll(Sequence(3,1,2), { Value : ThisRecord.Value, Value2 : ThisRecord.Value * ThisRecord.Value}), 
  LeftRecord.Value = RightRecord.Value, 
  JoinType.Left,
  RightRecord.Value2 As V2)


```

**Comment by MikeStall** [comment]
It's regressed with #2728 

You can repro just compiling `Join()`
Its's coming from
 https://github.com/microsoft/Power-Fx/blob/main/src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs#L4476

In syntax error cases, actual .Args is 0-length. This also happens while typing.

**Comment by MikeStall** [comment]
Fixed wiht #2758
